invenio_ldapclient/views.py

```diff
@@ -139,31 +139,34 @@ def _find_or_register_user(connection, username):
         return _register_or_update_user(entries)
 
 
-@blueprint.route('/ldap-login', methods=['POST'])
-def ldap_login_view():
-    """Process login request using LDAP and register the user if needed."""
+@blueprint.route('/ldap-login', methods=['GET', 'POST'])
```

Comment:
Maybe this is what your thumbs up below refers to, but let's be explicit so it is clear!

Both GET and POST requests to `/ldap-login` are served by `ldap_login()`. The code inside it takes this into account. Sidenote, this is a common Flask pattern - I am not doing something unique. The difference in functionality hinges on `form.validate_on_submit`. It is a `flask-wtform` convenient method that checks if the request is a POST first (`on_submit` part of the name) and then checks the validity of the form (`validate` part of the name). It will be true if the request is a POST, the CSRF token is valid and the form is valid.

Reply:
Sorry, the 👍 was for adding `form.validate_on_submit()`

I see, so for either 'GET' or 'POST' requests, the code in `ldap_login()` is executed? The other way just separates the two requests, so then I can see what is executed during which request.